### PR TITLE
new parameters for 1-octanol solvent (OPLSAA mixed model)

### PR DESCRIPTION
**Pull request by iorga**
- parameters from Zangi R (2018) Refinement of the OPLSAA force-field for liquid alcohols
  ACS Omega 3(12):18089–18099. doi: 10.1021/acsomega.8b03132
- see #120

### CODE REVIEW

File: mdpow/forcefields.py [hunk 1]
```diff
@@ -212,7 +224,12 @@ def get_solvent_model(identifier, forcefield='OPLS-AA'):
         return GROMACS_WATER_MODELS[identifier]
     except KeyError:
         try:
-            return GROMACS_SOLVENT_MODELS[forcefield][identifier]
+            if (forcefield=="OPLS-AA") and ((identifier=="octanol") or (identifier=="wetoctanol")):
+                if model==None:
+                    model = identifier
+                return GROMACS_SOLVENT_MODELS[forcefield][identifier][model]
```
Comment (VOD555) [inline mdpow/forcefields.py:230]: Use keyword `model` to select new or old octanol parameters.

File: mdpow/forcefields.py [hunk 2]
```diff
@@ -141,13 +141,25 @@ def get_water_model(watermodel=DEFAULT_WATER_MODEL):
         raise ValueError(msg)
 
 #: Other solvents (not water, see :data:`GROMACS_WATER_MODELS` for those).
-OPLS_SOLVENT_MODELS = {
+OPLS_OCTANOL_MODELS = {
```
Comment (VOD555) [inline mdpow/forcefields.py:144]: Add octanol dictionaries to instore new and old octanol parameter files.

File: mdpow/equil.py [hunk 3]
```diff
@@ -197,6 +197,7 @@ def __init__(self, molecule=None, **kwargs):
             self.solventmodel = forcefields.get_solvent_model(
                 self.solventmodel_identifier,
                 forcefield=forcefield,
+                model=solventmodel
```
Comment (VOD555) [inline mdpow/equil.py:200]: .Use keyword `solventmodel` to choose OPLS-AA octanol models. If one wants new parameters, should set `forcefield=OPLS-AA`, `solvent=octanol\wetoctanol` and `solventmodel=octanolnew\wetoctanolnew`

**Comment by orbeckst** [comment]
@VOD555 if you want to see changes, make your review "request changes" instead of just commenting. It's much better to be clear what needs to be done.

File: mdpow/equil.py (hunk 3 above)
Comment (orbeckst) [reply mdpow/equil.py:200]: Why are there backslashes in `octanol\wetoctanol`? I hope that this is actually `octanol/wetoctanol` or did you just mean to say "`octanol` _or_ `wetoctanol`?

File: mdpow/forcefields.py [hunk 4]
```diff
@@ -141,13 +141,25 @@ def get_water_model(watermodel=DEFAULT_WATER_MODEL):
...
-    'cyclohexane': GromacsSolventModel(
-        identifier="cyclohexane", itp="1cyclo.itp", coordinates="1cyclo.gro"),
+    'octanolnew': GromacsSolventModel(
+        identifier="octanol", itp="1octnew.itp", coordinates="1oct.gro")
+    }
+
+OPLS_WETOCTANOL_MODELS = {
     'wetoctanol': GromacsSolventModel(
         identifier="wetoctanol", itp="1octwet.itp", coordinates="1octwet.gro"),
+    'wetoctanolnew': GromacsSolventModel(
```
Comment (orbeckst) [inline mdpow/forcefields.py:154]: add a comment what "new" means, i.e., the citation for the parameters – "new" doesn't mean anything to anyone looking at this

File: mdpow/tests/test_forcefields.py [hunk 5]
```diff
@@ -152,22 +152,6 @@ def test_get_solvent_wetoctanol(forcefield):
         assert (mdpow.forcefields.get_solvent_model(model, forcefield=forcefield) is
                 mdpow.forcefields.GROMACS_SOLVENT_MODELS[forcefield][model])
 
-    @staticmethod
```
Comment (orbeckst) [inline mdpow/tests/test_forcefields.py:155]: Why were the tests removed?

File: mdpow/forcefields.py (hunk 4 above)
Comment (orbeckst) [reply mdpow/forcefields.py:154]: Zangi R (2018) Refinement of the OPLSAA force-field for liquid alcohols. 
; ACS Omega 3(12):18089–18099. doi: 10.1021/acsomega.8b03132

File: mdpow/forcefields.py (hunk 1 above)
Comment (VOD555) [reply mdpow/forcefields.py:230]: It's a comment.

File: mdpow/equil.py (hunk 3 above)
Comment (VOD555) [reply mdpow/equil.py:200]: That's `octanol` or `wetoctanol`

File: mdpow/tests/test_forcefields.py (hunk 5 above)
Comment (VOD555) [reply mdpow/tests/test_forcefields.py:155]: These tests do the same thing as the previous 3 tests.

File: mdpow/forcefields.py [hunk 6]
```diff
@@ -212,7 +224,12 @@ def get_solvent_model(identifier, forcefield='OPLS-AA'):
         return GROMACS_WATER_MODELS[identifier]
     except KeyError:
         try:
-            return GROMACS_SOLVENT_MODELS[forcefield][identifier]
+            if (forcefield=="OPLS-AA") and ((identifier=="octanol") or (identifier=="wetoctanol")):
+                if model==None:
+                    model = identifier
+                return GROMACS_SOLVENT_MODELS[forcefield][identifier][model]
+            else:
+                return GROMACS_SOLVENT_MODELS[forcefield][identifier]
```
[the remark this answers is not in the document]
Comment (VOD555) [reply mdpow/forcefields.py:232]: I think I've a idea. Maybe it's better to just make `GROMACS_SOLVENT_MODELS[forcefield][identifier]` to be `GROMACS_SOLVENT_MODELS[forcefield][model]`

File: mdpow/forcefields.py [hunk 7]
```diff
@@ -141,13 +141,25 @@ def get_water_model(watermodel=DEFAULT_WATER_MODEL):
...
         identifier="wetoctanol", itp="1octwet.itp", coordinates="1octwet.gro"),
+    'wetoctanolnew': GromacsSolventModel(
+        identifier="wetoctanol", itp="1octwetnew.itp", coordinates="1octwet.gro")
+    }
+
+OPLS_SOLVENT_MODELS = {
+    'octanol': OPLS_OCTANOL_MODELS,
+    'cyclohexane': GromacsSolventModel(
+        identifier="cyclohexane", itp="1cyclo.itp", coordinates="1cyclo.gro"),
+    'wetoctanol': OPLS_WETOCTANOL_MODELS,
```
[the remark this answers is not in the document]
Comment (VOD555) [reply mdpow/forcefields.py:162]: Yes, I think making model as identifier is a good choice. Currently, as identifiler is `solventmodel`, which is the keyword used to choose setup function, if I add a new `solventdmodel`, I have to add a new setup function.

File: mdpow/forcefields.py (hunk 4 above)
Comment (VOD555) [reply mdpow/forcefields.py:154]: I put the citation into ``description`` attribute of the class.

File: mdpow/forcefields.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (orbeckst) [reply mdpow/forcefields.py]: Ok, works for me. Then we would have things like
* "octanol/JorgensenXXX/w=27%"  for the old wet octanol... was it 27% water? (@iorga what would be the reference for our original octanol parameters, ie. what should JorgensenXXX be?)
* "octanol/ZangiYYYY/w=0%" (for the dry new octanol parameters)
* "water/TIP3P" 
* ...

File: mdpow/forcefields.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (orbeckst) [reply mdpow/forcefields.py]: Ok, see https://github.com/Becksteinlab/MDPOW/pull/121#discussion_r463352569

**Pull request merged**